github_actions/lib/dependabot/github_actions/update_checker.rb

```diff
@@ -82,6 +92,27 @@ def fetch_latest_version_for_git_dependency
         nil
       end
 
+      def fetch_lowest_security_fix_version
+        # TODO: Support Docker sources
+        return unless git_dependency?
+
+        fetch_lowest_security_fix_version_for_git_dependency
+      end
+
+      def fetch_lowest_security_fix_version_for_git_dependency
+        lowest_security_fix_version_tag.fetch(:version)
+      end
+
+      def lowest_security_fix_version_tag
+        @lowest_security_fix_version_tag ||= begin
+          relevant_tags = git_commit_checker.local_tags_for_allowed_versions
+
+          relevant_tags = Dependabot::UpdateCheckers::VersionFilters.filter_vulnerable_versions(relevant_tags,
```

Comment:
We may also filter out ignored versions here - might be best to check another ecosystem to confirm.

Reply:
Good point @Nishnha. Looking at other ecosystems, some filter out ignored versions, some don't. And also, some filter out lower versions, and some don't. Not sure about ignored versions, but for lower versions I believe the logic in those other ecosystems is wrong because even if it seems ok to make a best effort to not downgrade the dependency, we should still downgrade if strictly necessary, and I don't think other ecosystems are doing that... 🤔.

Anyways, I was thinking on punting on this now and maybe return to it later if issues are reported.

Reply:
when we were working on npm transitive dependency support we made it a point to filter out any lower versions to avoid downgrading dependencies - we would rather fail the job than downgrade. So I think we would want to do that here too

cc/ @mctofu

Reply:
Ok, so we prefer to leave the alert unresolved over proposing a downgrade, correct? I'm not sure I agree, but I'm happy to mimic that behavior! I think in the current state a downgrade would be proposed, so it needs some changes.

Reply:
Actually, I'm looking into it now and all ecosystems seem consistent here both in terms of ignored versions and not downgrading. I will update this.

Reply:
I reviewed this now. Ignored versions were already being ignored because that logic is actually built-in to `GitCommitChecker`.

Regarding lower versions, I actually learned that the logic to avoid downgrade PRs is actually very necessary because if there's both fixed versions higher and lower than the current version, we would be choosing a downgrade over an upgrade otherwise. I had actually missed this so that was a pretty bad bug.

I fixed the lower versions case and added specs for both ignored and lower versions 👍.